controllers/scaledjob_controller.go

```diff
@@ -80,47 +81,63 @@ func (r *ScaledJobReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, err
 	}
 
-	var errMsg string
-	if scaledJob.Spec.JobTargetRef != nil {
-		reqLogger.Info("Detected ScaleType = Job")
-		conditions := scaledJob.Status.Conditions.DeepCopy()
-		msg, err := r.reconcileScaledJob(reqLogger, scaledJob)
-		if err != nil {
-			reqLogger.Error(err, msg)
-			conditions.SetReadyCondition(metav1.ConditionFalse, "ScaledJobCheckFailed", msg)
-			conditions.SetActiveCondition(metav1.ConditionUnknown, "UnknownState", "ScaledJob check failed")
-			r.Recorder.Event(scaledJob, corev1.EventTypeWarning, eventreason.ScaledJobCheckFailed, msg)
-		} else {
-			wasReady := conditions.GetReadyCondition()
-			if wasReady.IsFalse() || wasReady.IsUnknown() {
-				r.Recorder.Event(scaledJob, corev1.EventTypeNormal, eventreason.ScaledJobReady, "ScaledJob is ready for scaling")
-			}
-			reqLogger.V(1).Info(msg)
-			conditions.SetReadyCondition(metav1.ConditionTrue, "ScaledJobReady", msg)
+	// ensure Status Conditions are initialized
+	if !scaledJob.Status.Conditions.AreInitialized() {
+		conditions := kedav1alpha1.GetInitializedConditions()
+		if err := kedacontrollerutil.SetStatusConditions(r.Client, reqLogger, scaledJob, conditions); err != nil {
+			return ctrl.Result{}, err
 		}
+	}
 
+	// Check jobTargetRef is specified
+	if scaledJob.Spec.JobTargetRef == nil {
+		errMsg := "scaledJob.spec.jobTargetRef is not set"
+		err := fmt.Errorf(errMsg)
+		reqLogger.Error(err, "scaledJob.spec.jobTargetRef not found")
 		return ctrl.Result{}, err
 	}
+	msg, err := r.reconcileScaledJob(reqLogger, scaledJob)
+	conditions := scaledJob.Status.Conditions.DeepCopy()
+	if err != nil {
+		reqLogger.Error(err, msg)
+		conditions.SetReadyCondition(metav1.ConditionFalse, "ScaledJobCheckFailed", msg)
+		conditions.SetActiveCondition(metav1.ConditionUnknown, "UnknownState", "ScaledJob check failed")
+		r.Recorder.Event(scaledJob, corev1.EventTypeWarning, eventreason.ScaledJobCheckFailed, msg)
+	} else {
+		wasReady := conditions.GetReadyCondition()
+		if wasReady.IsFalse() || wasReady.IsUnknown() {
+			r.Recorder.Event(scaledJob, corev1.EventTypeNormal, eventreason.ScaledJobReady, "ScaledJob is ready for scaling")
+		}
+		reqLogger.V(1).Info(msg)
+		conditions.SetReadyCondition(metav1.ConditionTrue, "ScaledJobReady", msg)
+	}
 
-	errMsg = "scaledJob.Spec.JobTargetRef is not set"
-	err = fmt.Errorf(errMsg)
-	reqLogger.Error(err, "scaledJob.Spec.JobTargetRef not found")
+	if err := kedacontrollerutil.SetStatusConditions(r.Client, reqLogger, scaledJob, &conditions); err != nil {
+		return ctrl.Result{}, err
+	}
 	return ctrl.Result{}, err
 }
 
-// reconcileJobType implemets reconciler logic for K8s Jobs based ScaleObject
+// reconcileJobType implemets reconciler logic for K8s Jobs based ScaledJob
```

Comment:
```suggestion
// reconcileScaledJob implemets reconciler logic for K8s Jobs based ScaledJob
```

Reply:
Thanks, @zroubalik for taking a look at this.
I will do update this also.
one more thing Should I also update  Changelog?

Reply:
Hi  @zroubalik 
Addressed the comment, PTAL

Reply:
> Looking good, minor nit and could you please add this to the changelog? (Improvements section)
Sorry I missed your comment

I updated the changelog. PTAL